### PR TITLE
add an extension for bfloat16 conversions

### LINKED ISSUE #14: bfloat16 conversions extension
Add an extension to indicate formal support for bfloat16 conversions.  There will be no new APIs added by this extension and it will only update the SPIR-V environment to indicate that bfloat16 conversions are supported.

See: https://github.com/intel/llvm/blob/sycl/sycl/doc/design/spirv-extensions/SPV_INTEL_bfloat16_conversion.asciidoc

See also: https://github.com/KhronosGroup/OpenCL-Docs/pull/762

### PR DESCRIPTION
**Pull request by bashbaug**
fixes #14

Adds the ZE_extension_bfloat16_conversions extension.

This extension adds no new APIs but does allow usage of bfloat16 conversion instructions in SPIR-V modules.

### CODE REVIEW

**Review by jandres742 — APPROVED** [review]

File: scripts/core/bfloat16conversions.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#
+# Copyright (C) 2021-2022 Intel Corporation
```
Comment (jandres742) [inline scripts/core/bfloat16conversions.yml:2]: 2022 only.

File: scripts/core/SPIRV.rst [hunk 2]
```diff
@@ -665,6 +665,27 @@ include the **LinkOnceODR** linkage type.
...
+
+When use of the ``SPV_INTEL_bloat16_conversion`` extension is declared in the
+module via **OpExtension**, the environment must accept modules that
+declare the **Bfloat16ConversionINTEL** capability.
+
+For the instructions **OpConvertFToBF16INTEL** and **OpConvertBF16ToFINTEL**
+added by the extension:
+
+- Valid types for *Result Type*, *Float Value*, and *Bfloat16 Value* are Scalars
+  and **OpTypeVectors** with 2, 3, 4, 8, or 16 *Component Count* components
```
Comment (jandres742) [inline scripts/core/SPIRV.rst:685]: are we missing period at the end?

File: scripts/core/bfloat16conversions.yml (hunk 1 above)
Comment (bashbaug) [reply scripts/core/bfloat16conversions.yml]: fixed

**Review by jandres742 — APPROVED** [review]

File: scripts/core/SPIRV.rst (hunk 2 above)
Comment (bashbaug) [reply scripts/core/SPIRV.rst:685]: I don't think so - both here and elsewhere in the doc we consder these list elements, hence no period required.

**Pull request merged**